### DISCUSSION TITLE
Bticino's OTA updates not work

### DISCUSSION THREAD

**Discussion by leoogermenia** [Koenkk/zigbee2mqtt#20780] [Devices]
## What happened?

I'm trying to update my switch (K4003C), but I recived multiple errors. Sometimes I can start the update, but when it arrives to 98/99% it fails

## Zigbee2MQTT version
1.35.1
## Adapter firmware version
6.10.3.0 build 297
## Adapter
ITead Sonoff Zigbee 3.0 USB Dongle Plus V2 model "ZBDongle-E"
## Setup
Add-on on HA (Proxmox VM)

## Debug Log

`Update of 'Lampadario Cameretta' failed (OTA: Timeout, device did not request any image blocks)`

**Comment by Krewwell** [comment]
I have the same provblem but no solution for now....maybe someone else can help :(

**Comment by FabianMangold** [comment]
Hi again @leoogermenia and @Krewwell ,

Any chance you could give the update a try using the latest DEV versions of Z2m please ?
Several features were implemented to improve the Legrand / BTicino OTA updates. Despite those, you may need to power cycle the devices prior to the upgrade still, and give the OTA several attempts in a row, but this should do the trick.

Thanks for any feedback you may provide.

**Reply by leoogermenia** [reply]
Hi again @FabianMangold 

At the end I was able to update all the device simply trying lots of time to update it and after some errors they can finally update it.

**Reply by Krewwell** [reply]
Hi @FabianMangold thank you! I have seen the last release of z2m and the new features for legrand/bticino devices. 
Thanks to you, now i have update 6 devices BTicino but i had to try several time before I succeeded. I still have two more devices to update (of the same type as those already updated) but after several attempts I still haven't succeeded. I've already tried turning off the power for a few minutes but the update doesn't start. In any case I will try again (if I managed to update the others I will succeed for these too).

Have a nice day and thanks again!

**Reply by leoogermenia** [reply]
@Krewwell ciao, quali dispositivi hai che non ti si aggiornano?

[Translate]
Hi, what devices aren't updating?

**Reply by Krewwell** [reply]
Hi @leoogermenia. I use the K4027C and the k4003C BTicino. The K4027C is the most difficult to upgrade

**Reply by leoogermenia** [reply]
@Krewwell i was able to update it with lots of tries.. I have only one K4027C and it update with "no" problem

**Comment by FabianMangold** [comment]
Thanks both.

@leoogermenia, can we close this issue ?
@Krewwell: Ok perfect, let me know if you still have problems. If so, please open a new issue, and just tag me so that we can investigate.
